src/middlewared/middlewared/etc_files/krb5.conf

```diff
@@ -0,0 +1,153 @@
+                        section_conf.update({default_section: {binding[0].strip(): binding[1].strip()}})
+                else:
+                    continue
+
+            return section_conf
+
+        db = {}
+        db['ad'] = middleware.call_sync('datastore.query', 'directoryservice.activedirectory', None, {'get': True})
+        db['ldap'] = middleware.call_sync('datastore.query', 'directoryservice.ldap', None, {'get': True})
+        db['cifs'] = middleware.call_sync('datastore.query', 'services.cifs', None, {'get': True})
+        db['krb_aux'] = middleware.call_sync('datastore.query', 'directoryservice.kerberossettings', None, {'get': True})
+        db_realms = middleware.call_sync('datastore.query', 'directoryservice.kerberosrealm')
+        appdefaults = {'pam': {'forwardable': 'true', 'ticket_lifetime': '86400', 'renew_lifetime': '86400'}}
+
+        libdefaults = {
+                        'krb5_main': {
+                                      'dns_lookup_realm': 'true',
+                                      'dns_lookup_kdc': 'true',
+                                      'ticket_lifetime': '24h',
+                                      'clockskew': '300',
+                                      'forwadable': 'yes'
+                                     }
+                       }
+
+        if db['ad']['ad_enable'] and db['ad']['ad_kerberos_realm']['krb_realm']:
+            krb_default_realm = db['ad']['ad_kerberos_realm']['krb_realm']
+        elif db['ad']['ad_enable']:
+            krb_default_realm = db['ad']['ad_domainname']
+        elif db['ldap']['ldap_enable'] and db['ldap']['ldap_kerberos_realm']:
+            krb_default_realm = db['ldap']['ldap_kerberos_realm']
+        else:
+            krb_default_realm = None
+
+        if krb_default_realm:
+            libdefaults['krb5_main'].update({'default_realm': krb_default_realm})
+
+        parsed_appdefaults = parse_defaults("appdefault", appdefaults, db_def=db['krb_aux']['ks_appdefaults_aux'])
+        parsed_libdefaults = parse_defaults("libdefault", libdefaults, db_def=db['krb_aux']['ks_libdefaults_aux'])
+        db_realms = middleware.call_sync('datastore.query', 'directoryservice.kerberosrealm')
+
+%>
+[app_defaults]
+% for section_name, section in parsed_appdefaults.items():
+            % if section_name == "krb5_main":
+            % for binding, value in section.items():
+            ${binding} = ${value}
+            % endfor
+            % else:
+            ${section_name} = {
+            % for binding, value in section.items():
+                   ${binding} = ${value}
+            % endfor
+            }
+            % endif
+% endfor
+
+[lib_defaults]
+% for section_name, section in parsed_libdefaults.items():
+            % for binding, value in section.items():
+            ${binding} = ${value}
+            % endfor
+% endfor
+
+[domain_realm]
+% for realm in db_realms:
+            ${f'{realm["krb_realm"]}'.lower()} = ${f'{realm["krb_realm"]}'}
```

Comment:
I am confused here, why do you need f string?

Just ${realm["krb_realm"]} does not work?